bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSNodeHandler.java

```diff
@@ -584,7 +620,7 @@ private void initializeChannelAndConfigState(Node node, ZwaveJSTypeGeneratorResu
         }
     }
 
-    private void detectColorCapabilities(Node node) {
+    private ThingBuilder detectColorCapabilities(ThingBuilder builder, Node node) {
```

Comment:
Looking at your proposal, i see how you want to fix this and that looks great. Besides the previous comment (a nit) my main concern is that this method no longer detects capability it also adds channels.

Would it be possible to move the channel creation in the `ZwaveJSTypeGeneratorResult` as that class is responsible for the channels created. It would be hard to maintain a binding that has multiple places whwere channels are created and it would also imrpove testability.

Reply:
It’s complicated. One needs to have all the native channels before one can detect the capability. And one needs the capability before one can create the extra CT channel. A chicken and egg situation.

Reply:
Ok